core/state/statedb.go

```diff
@@ -876,7 +876,11 @@ func (s *StateDB) IntermediateRoot(deleteEmptyObjects bool) common.Hash {
 	// Now we're about to start to write changes to the trie. The trie is so far
 	// _untouched_. We can check with the prefetcher, if it can give us a trie
 	// which has the same root, but also has some content loaded into it.
-	if prefetcher != nil {
+	//
+	// Don't check prefetcher if Verkle Trie has been used. In the context of Verkle,
+	// only a single trie is used for state hashing. Replacing a non-nil Verkle tree
+	// here could result in losing uncommitted changes from storage.
+	if prefetcher != nil && (s.trie == nil || !s.trie.IsVerkle()) {
```

Comment:
This change does not only affect verkle, but also non-verkle, if `prefetcher != nil, s.trie != nil` would previously enter  this clause, but not after this change.  
Please elaborate

Reply:
`s.trie != nil && !s.trie.IsVerkle()` will still entry the clause
only `s.trie != nil && s.trie.IsVerkle` will be rejected

Reply:
I'm talking about non-verkle (so `s.trie.IsVerkle() == false`). 

So, for `prefetcher != nil, s.trie != nil`
-  _currently_, on `master`: we enter the clause 
-  in this PR: we do not enter the clause

Reply:
for the case you mentioned, non-verkle, trie is not nil, we will still enter the clause isn't it?

`(s.trie == nil || !s.trie.IsVerkle())`, the case will satisfy the second condition `!s.trie.IsVerkle()`

Reply:
Oh right, yes, my bad!